curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/american-british-translator.md

```diff
@@ -28,28 +38,225 @@ tests:
-    testString: ''
-  - text: All 4 functional tests are complete and passing. See <code>/tests/2_functional-tests.js</code> for the functionality you should write tests for.
-    testString: ''
+
+  - text: I can <code>POST</code> to <code>/api/translate</code> with a body containing <code>text</code> with the text to translate and <code>locale</code> with either <code>american-to-british</code> or <code>british-to-american</code>, The returned object should contain the submitted <code>text</code> and <code>translation</code> with the translated text.  
+    testString: "async getUserInput => {
+        try {
+          const text = 'Mangoes are my favorite fruit.';
+          const locale = 'american-to-british';
+          const output = {
+            text: 'Mangoes are my favorite fruit.',
+            translation: 'Mangoes are my <span class=\"highlight\">favourite</span> fruit.'
+          };
+          let data = await fetch(getUserInput('url') + '/api/translate', {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({text, locale})
+          });
+          let parsed = await data.json();
+          assert.isObject(parsed);
+          assert.property(parsed, 'text');
+          assert.property(parsed, 'translation');
+          assert.deepEqual(parsed, output);        
+        } catch (err) {
+          throw new Error(err.responseText || err.message);
+        }
+      }"
+  - text: The <code>/api/translate</code> route should handle the way time is written in American and British English. For example, ten thirty is written as "10.30" in British English and "10:30" in American English.  
+    testString: "async getUserInput => {
+      try {
+        const text = 'Lunch is at 12:15 today.';
+        const locale = 'american-to-british';
+        const output = {
+          text: text,
+          translation: 'Lunch is at <span class=\"highlight\">12.15</span> today.'
+        };
+        let data = await fetch(getUserInput('url') + '/api/translate', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({text, locale})
+        });
+        let parsed = await data.json();
+        assert.isObject(parsed);
+        assert.property(parsed, 'text');
+        assert.property(parsed, 'translation');
+        assert.deepEqual(parsed, output);
+      } catch (err) {
+        throw new Error(err.responseText || err.message);
+      }
+    }"
+  - text: The <code>/api/translate</code> route should also handle the way titles/honorifics are abbreviated in American and British English. For example, Doctor Wright is abbreviated as "Dr Wright" in British English and "Dr. Wright" in American English. See <code>/public/american-to-british-titles.js</code> for the different titles your application should handle.
```

Comment:
I worry that when we say "Doctor Wright is abbreviated..." that users will translate "doctor" to `Dr/Dr.` - Is that something the app is supposed to do? Similar issue with the test above this one. Perhaps a slight reword... Maybe, "For example, `Dr Wright` in British English would be `Dr. Wright` in American English.

Perhaps it's not a big deal if they do or don't translate that, not sure if there's any tests for it.

Reply:
I didn't have any issues with this, considering that it's followed by the reference to the `american-to-british-titles.js` which clearly spells out the translations.  We don't have any tests for "Doctor" that I am aware of, so it seems like a moot point.

Reply:
Yea, I'm fine with it then.